apps/web/app/routes/examples/actions/additional-input.tsx

```diff
@@ -0,0 +1,70 @@
+import hljs from 'highlight.js/lib/common'
+import type {
+  ActionFunction,
+  LoaderFunction,
+  MetaFunction,
+} from '@remix-run/node'
+import { formAction } from '~/formAction'
+import { z } from 'zod'
+import Form from '~/ui/form'
+import { metaTags } from '~/helpers'
+import { makeDomainFunction } from 'domain-functions'
+import Example from '~/ui/example'
+
+const title = 'Additional input'
+const description =
+  'In this example, we use different schemas for the form and the mutation, passing additional mutation input to formAction.'
+
+export const meta: MetaFunction = () => metaTags({ title, description })
+
+const code = `const formSchema = z.object({
+  firstName: z.string().min(1),
+  email: z.string().min(1).email(),
+})
+
+const mutationSchema = formSchema.extend({
+  country: z.enum(['BR', 'US']),
+})
+
+const mutation = makeDomainFunction(mutationSchema)(async (values) => values)
+
+export const action: ActionFunction = async ({ request }) =>
```

Comment:
Shouldn't we use `ActionArgs` here for better type inference?

Reply:
Only if we want to promote Remix + TS best practices on the docs, which I think might be valid. But we don't need it with Remix Forms, because Remix Forms is the one that calls `useActionData` and makes sure the typing is correct. We'll almost never call `useActionData<typeof action>` when using Remix Forms.

For now, I've been preferring the shorter code that arrow functions give us. I already feel there is a lot of code to read in the examples, so I take every opportunity I have to have fewer lines of code :)

But I'm very open to changing this if you really care about promoting better Remix TS practices on the docs.

Reply:
Good point, ship it